apps/openmw/mwmechanics/character.cpp

```diff
@@ -697,7 +697,12 @@ CharacterController::CharacterController(const MWWorld::Ptr &ptr, MWRender::Anim
             mAnimation->showCarriedLeft(updateCarriedLeftVisible(mWeaponType));
         }
 
-        if(!cls.getCreatureStats(mPtr).isDead())
+        if(!cls.getCreatureStats(mPtr).isDead() ||
+		(cls.isNpc() &&
+		 mPtr.get<ESM::NPC>()->mBase->mFlags & 8)) // persistent !
```

Comment:
Done, so finally no more bit test. The problem was that the character was
killed in CharacterController::CharacterController even before any attached
script has a chance to stop the animation.
So I just removed completely the test isDead from here, it will be killed
later anyway.
Checked on both mods, everything works. And went to a cell with a dead
body, it stayed dead.
This time I think everything is ok (finally !)


2015-09-15 18:59 GMT+02:00 Emmanuel Anne <emmanuel.anne@gmail.com>:

> Yeah I didn't interpret the description there correctly, the only thing
> persist does is that it just prevents them from disappearing after a while.
> And this means I was too fast and too optimistic about that, this bit is
> probably unrelated.
> Which means that I will have to look again how these mannequins behave
> (the 1st kind, the one from "a good place to stay").
> Ok, I'll post again later about that, but I won't do it just now...
>
> 2015-09-15 14:26 GMT+02:00 scrawl <notifications@github.com>:
>
>> In apps/openmw/mwmechanics/character.cpp
>> <https://github.com/OpenMW/openmw/pull/759#discussion_r39503407>:
>>
>> > @@ -697,7 +697,12 @@ CharacterController::CharacterController(const MWWorld::Ptr &ptr, MWRender::Anim
>> >              mAnimation->showCarriedLeft(updateCarriedLeftVisible(mWeaponType));
>> >          }
>> >
>> > -        if(!cls.getCreatureStats(mPtr).isDead())
>> > +        if(!cls.getCreatureStats(mPtr).isDead() ||
>> > +		(cls.isNpc() &&
>> > +		 mPtr.get<ESM::NPC>()->mBase->mFlags & 8)) // persistent !
>>
>> If you have found a new flag, the flag should be added to the enum in
>> ESM::Npc, to esmtool and OpenCS. I kinda doubt this is correct, though. We
>> already have bool mPersistent in ESM::Npc set based on the record flags.
>> And what does persist have to do with mannequins?
>>
>> From
>> http://wiki.theassimilationlab.com/mmw/Morrowind_Script_Library#Mannequins
>> mannequins are npc with the persist bit, they don't play the
>> death animation and just stay here
>>
>> I would guess they don't play death animations because of the skipAnim
>> call, the Persist bit should have nothing to do with it.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/OpenMW/openmw/pull/759/files#r39503407>.
>>
>
>